crates/sui-types/src/storage.rs

```diff
@@ -59,12 +59,10 @@ pub struct SingleTxContext {
 }
 
 impl SingleTxContext {
+    // legacy
     pub fn transfer_sui(sender: SuiAddress) -> Self {
         Self::sui_transaction(ident_str!("transfer_sui"), sender)
```

Comment:
@patrickkuo I was fairly surprised by this setup for Events, it feels very brittle as these modules do not exist. Thoughts on something more structured? e.g. 
```
pub enum EventLocation {
    // System
    Genesis,
    // legacy
    TransferSui,
    Pay,
    PaySui,
    PayAllSui,
    // programmable transaction command
    SplitCoin,
    // used both in legacy and programmable transactions
    TransferObject,
    UnusedInputObject,
    Gas,
    ModuleInit,
    MoveCall {
        package: ObjectID,
        module: Identifier,
    },
}
```

Reply:
these are used to fill the gaps between system event and user event, since we have decided to remove them completely, we shouldn't need these anymore

Reply:
Cool! Is there an issue or PR to track the removal?